### DISCUSSION TITLE
Question about TouchEffect - moving from Xamarin

### DISCUSSION THREAD

**Discussion by ziomek64** [CommunityToolkit/Maui#361] [Q&A] [answered]
I'm trying to move from Xamarin and that toolkit. There is no TouchEffect here? I need TouchEffect.Nativeanimation and Toucheffect.command

**Comment by TheCodeTraveler** [comment, marked as answer]
Hi @ziomek64!

To use `TouchEffect` with your .NET MAUI app today, we recommend using the [`Xamarin.CommunityToolkit.MauiCompat` NuGet Package](https://www.nuget.org/packages/Xamarin.CommunityToolkit.MauiCompat).

`Xamarin.CommunityToolkit.MauiCompat` contains all of the same features as `Xamarin.CommunityToolkit`, ported to .NET MAUI. You can read more information about it here: [https://devblogs.microsoft.com/xamarin/introducing-net-maui-compatibility-for-the-xamarin-community-toolkit/](https://devblogs.microsoft.com/xamarin/introducing-net-maui-compatibility-for-the-xamarin-community-toolkit//?WT.mc_id=mobile-39516-bramin).

Long story short, Effects are no longer recommended in .NET MAUI. We plan to introduce the functionality of `TouchEffect` from `Xamarin.CommunityToolkit` as a `Behavior` in the .NET MAUI Community Toolkit, and you can follow the progress here:
- https://github.com/CommunityToolkit/Maui/issues/124
- https://github.com/CommunityToolkit/Maui/issues/86

We would love your help writing the code for the `Touch` behavior! The Community Toolkit contains features for, and created by, the .NET MAUI Community. We are all volunteers, contributing code in our spare time on nights and weekends. If you'd like to contribute, you can learn more here: [https://devblogs.microsoft.com/dotnet/contributing-to-net-maui-community-toolkit/](https://devblogs.microsoft.com/dotnet/contributing-to-net-maui-community-toolkit/?WT.mc_id=mobile-0000-bramin)

**Comment by ziomek64** [comment]
@brminnick Should I use this package alongside maui toolkit or instead of it? Or it doesn't really matter?

**Reply by TheCodeTraveler** [reply]
When porting Xamarin.Forms apps to .NET MAUI, we recommend using `Xamarin.CommunityToolkit.MauiCompat` as a temporary bridge. Eventually, you'll want to use `CommunityToolkit.Maui` because `Xamarin.CommunityToolkit.MauiCompat` is not receiving any new features will be deprecated alongside Xamarin.Forms.

<a href="https://devblogs.microsoft.com/xamarin/introducing-net-maui-compatibility-for-the-xamarin-community-toolkit//?WT.mc_id=mobile-39516-bramin"><img width="784" alt="image" src="https://user-images.githubusercontent.com/13558917/163626726-16d8257d-5306-4c6f-84ba-8161076be1ef.png"></a>

**Reply by ziomek64** [reply]
ok @brminnick  so I installed and trying to configure it MauiProgram.cs:
```
using Microsoft.Maui.Controls.Compatibility.Hosting;

namespace MauiApp1;
public static class MauiProgram
{

	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
            .ConfigureMauiHandlers(handlers =>
            {
                // Register ALL handlers in the Xamarin Community Toolkit assembly
                handlers.AddCompatibilityRenderers(typeof(Xamarin.CommunityToolkit.UI.Views.MediaElementRenderer).Assembly);
            })
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			});
            

        return builder.Build();
	}
}
```
I'm getting error that MediaElementRenderer does not exist in the Xamarin.CommunityToolkit.UI.Views

**Reply by TheCodeTraveler** [reply]
Could you provide the stack trace? 

Is the error happening on iOS, Android or Windows? `Xamarin.CommunityToolkit.MauiCompat` doesn't yet support Windows.

**Reply by ziomek64** [reply]
@brminnick that explains a lot because I was trying to run the app on Windows. Android on the other hand throws this when the nuget package is installed. Stangely uninstalling the nuget does not fix it. The app was working on android before nuget was installed.
`java.lang.RuntimeException: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: C:\Users\mrady\.nuget\packages\xamarin.google.guava.listenablefuture\1.0.0.5\buildTransitive\net6.0-android31.0\..\..\jar\guava-listenablefuture.jar : com/google/common/util/concurrent/ListenableFuture.class	MauiApp2	C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\32.0.300-rc.1.4\tools\Xamarin.Android.D8.targets	79	
`

**Comment by CliffAgius** [comment]
Closed as answered
